src/System.Security.Cryptography.Cng/src/Internal/Cryptography/Helpers.cs

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics;
+using System.Globalization;
+using System.Security.Cryptography;
+using System.Runtime.InteropServices;
+
+using Microsoft.Win32.SafeHandles;
+
+using ErrorCode = Interop.NCrypt.ErrorCode;
+
+namespace Internal.Cryptography
+{
+    internal static class Helpers
+    {
+        public static byte[] CloneByteArray(this byte[] src)
+        {
+            return (byte[])(src.Clone());
+        }
+
+        //
+        // The C# construct
+        //   
+        //    fixed (byte* p = new byte[0])
```

Comment:
I'm going to offline this one. Maybe there's a cleanness issue here, maybe not but if there is, we want to look at this more holistically (i.e. X509Certificates also has this pattern) rather than throwing in patches into this PR.